internal/core/command/v2/application/command.go

```diff
@@ -113,6 +98,48 @@ func CommandsByDeviceName(name string, dic *di.Container) (deviceCoreCommand dto
 	return deviceCoreCommand, nil
 }
 
+func commandPath(deviceName, cmdName string) string {
+	return fmt.Sprintf("%s/%s/%s/%s", V2Routes.ApiDeviceRoute, V2Routes.Name, deviceName, cmdName)
+}
+func buildCoreCommand(deviceName, serviceUrl, cmdName, readWrite string) dtos.CoreCommand {
+	cmd := dtos.CoreCommand{
+		Name: cmdName,
+		Url:  serviceUrl,
+		Path: commandPath(deviceName, cmdName),
+	}
+	if strings.Contains(readWrite, V2Routes.ReadWrite_R) {
+		cmd.Get = true
+	}
+	if strings.Contains(readWrite, V2Routes.ReadWrite_W) {
+		cmd.Set = true
+	}
+	return cmd
+}
+
+func BuildCoreCommands(deviceName string, serviceUrl string, profile dtos.DeviceProfile) []dtos.CoreCommand {
```

Comment:
This shouldn't be externalized just for unit test purposes, but I do understand the need for code reuse... Can't you use `CommandsByDeviceName` and mock the DIC and DB to avoid making this external?

Reply:
No problem, I've got your point. Modified the test to make the function internalize,